ScriptFileWatcher does not cancel scheduled jobs

When uninstalling the scripted rule support, the following exception is thrown:

```
16:25:26.849 [ERROR] [automation.module.script.rulesupport] - FrameworkEvent ERROR - org.eclipse.smarthome.automation.module.script.rulesupport
java.io.IOException: Exception in opening zip file: /Users/kai/Downloads/oh/userdata/cache/org.eclipse.osgi/198/0/bundleFile
	at org.eclipse.osgi.framework.util.SecureAction.getZipFile(SecureAction.java:305)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.storage.bundlefile.ZipBundleFile.basicOpen(ZipBundleFile.java:85)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.storage.bundlefile.ZipBundleFile.getZipFile(ZipBundleFile.java:98)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.storage.bundlefile.ZipBundleFile.checkedOpen(ZipBundleFile.java:65)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.storage.bundlefile.ZipBundleFile.getEntry(ZipBundleFile.java:232)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.internal.loader.classpath.ClasspathManager.findClassImpl(ClasspathManager.java:562)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.internal.loader.classpath.ClasspathManager.findLocalClassImpl(ClasspathManager.java:540)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.internal.loader.classpath.ClasspathManager.findLocalClass(ClasspathManager.java:527)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.findLocalClass(ModuleClassLoader.java:324)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.internal.loader.BundleLoader.findLocalClass(BundleLoader.java:327)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.internal.loader.BundleLoader.findClassInternal(BundleLoader.java:402)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:352)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:344)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:160)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)[:1.8.0_20]
	at org.eclipse.smarthome.automation.module.script.rulesupport.internal.loader.ScriptFileWatcher.checkFiles(ScriptFileWatcher.java:225)[198:org.eclipse.smarthome.automation.module.script.rulesupport:0.9.0.201705120951]
	at org.eclipse.smarthome.automation.module.script.rulesupport.internal.loader.ScriptFileWatcher$$Lambda$12/1692515238.run(Unknown Source)[198:org.eclipse.smarthome.automation.module.script.rulesupport:0.9.0.201705120951]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)[:1.8.0_20]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)[:1.8.0_20]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)[:1.8.0_20]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)[:1.8.0_20]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)[:1.8.0_20]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)[:1.8.0_20]
	at java.lang.Thread.run(Thread.java:745)[:1.8.0_20]
Caused by: java.io.FileNotFoundException: /Users/kai/Downloads/oh/userdata/cache/org.eclipse.osgi/198/0/bundleFile (No such file or directory)
	at java.util.zip.ZipFile.open(Native Method)[:1.8.0_20]
	at java.util.zip.ZipFile.<init>(ZipFile.java:220)[:1.8.0_20]
	at java.util.zip.ZipFile.<init>(ZipFile.java:150)[:1.8.0_20]
	at java.util.zip.ZipFile.<init>(ZipFile.java:164)[:1.8.0_20]
	at org.eclipse.osgi.framework.util.SecureAction.getZipFile(SecureAction.java:288)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	... 23 more
```

--- a/bundles/automation/org.eclipse.smarthome.automation.module.script.rulesupport/src/main/java/org/eclipse/smarthome/automation/module/script/rulesupport/internal/loader/ScriptFileWatcher.java
+++ b/bundles/automation/org.eclipse.smarthome.automation.module.script.rulesupport/src/main/java/org/eclipse/smarthome/automation/module/script/rulesupport/internal/loader/ScriptFileWatcher.java
@@ -43,6 +43,7 @@ public class ScriptFileWatcher extends AbstractWatchService {
     private long earliestStart = System.currentTimeMillis() + INITIAL_DELAY * 1000;
 
     private ScriptEngineManager manager;
+    ScheduledExecutorService scheduler;
 
     private Map<String, Set<URL>> urlsByScriptExtension = new ConcurrentHashMap<>();
     private Set<URL> loaded = new HashSet<>();
@@ -59,7 +60,17 @@ public class ScriptFileWatcher extends AbstractWatchService {
     public void activate() {
         super.activate();
         importResources(new File(pathToWatch));
-        startScheduler();
+        scheduler = Executors.newSingleThreadScheduledExecutor();
+        scheduler.scheduleWithFixedDelay(this::checkFiles, INITIAL_DELAY, RECHECK_INTERVAL, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void deactivate() {
+        if (scheduler != null) {
+            scheduler.shutdownNow();
+            scheduler = null;
+        }
+        super.deactivate();
     }
 
     /**
@@ -214,11 +225,6 @@ public class ScriptFileWatcher extends AbstractWatchService {
 
     private String getScriptIdentifier(URL url) {
         return url.toString();
-    }
-
-    private void startScheduler() {
-        ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
-        scheduler.scheduleWithFixedDelay(this::checkFiles, INITIAL_DELAY, RECHECK_INTERVAL, TimeUnit.SECONDS);
     }
 
     private void checkFiles() {

